lib/puppet_x/puppetlabs/azure/provider_arm.rb

```diff
@@ -107,9 +107,15 @@ def start_vm(machine)
 
         def get_all_vms # rubocop:disable Metrics/AbcSize
           begin
-            vms = ProviderArm.compute_client.virtual_machines.list_all.value!.body.value
+            vms = []
+            result = ProviderArm.compute_client.virtual_machines.list_all
+            vms += result.value
+            while ! result.next_link.nil? and ! result.next_link.empty? do
```

Comment:
Hmm. Is this loop here to get nested VMs? What does `.next_link` provide? Do we have to think about a recursive solution to this?